discord/member.py

```diff
@@ -473,13 +436,24 @@ def _update(self, data: GuildMemberUpdateEvent) -> None:
         self._flags = data.get('flags', 0)
         self._avatar_decoration_data = data.get('avatar_decoration_data')
 
-    def _presence_update(self, data: PartialPresenceUpdate, user: UserPayload) -> Optional[Tuple[User, User]]:
+    def _presence_update(
+        self, data: PartialPresenceUpdate, raw: RawPresenceUpdateEvent, user: UserPayload
+    ) -> Optional[Tuple[User, User]]:
+        if raw._activities is None:
+            raw._create_activities(data, self._state)
+
+        self.activities = raw.activities
+        self.client_status = raw.client_status
+
+        if len(user) > 1:
+            return self._update_inner_user(user)
+
+    def _perf_presence_update(self, data: PartialPresenceUpdate, user: UserPayload) -> Optional[Tuple[User, User]]:
```

Comment:
Please revert the name.

I'm also not sure I understand the changeset with `raw` and `_activities` to require the change in the first place.